### PR TITLE
Redis polymorphism: Add redis store wrapper

### LINKED ISSUE #24: Redis adapter polimorphism
It should accept more connection providers.
Take a look: https://github.com/symfony/lock/blob/master/Store/RedisStore.php

### PR DESCRIPTION
**Pull request by b1rdex**
Closes #24 and closes #25

### CODE REVIEW

**Comment by b1rdex** [comment]
I added RedisStore wrapper to separate redis operations from actual Ganesha adapter implementation.
Need to add implementations for desired redis clients to RedisStore.

**Comment by b1rdex** [comment]
Please review. If this is acceptable than I can move on and add other redis clients implementations support.

**Comment by ackintosh** [comment]
Thanks for this PR! Looks good to me.

**Comment by b1rdex** [comment]
Hi @ackintosh. I finally made this green. Ready for review.

**Comment by ackintosh** [comment]
Thanks! I'll review this later today.

File: tests/Ackintosh/Ganesha/Storage/Adapter/PredisRedisTest.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class PredisRedisTest extends AbstractRedisTest
+{
+    /**
+     * @return \Predis\Client
+     */
+    protected function getRedisConnection()
+    {
+        $r = new \Predis\Client(getenv('GANESHA_EXAMPLE_REDIS') ?: 'localhost');
+        $r->connect();
```
Comment (ackintosh) [inline tests/Ackintosh/Ganesha/Storage/Adapter/PredisRedisTest.php]: I've got the error when run the tests in docker container. 👀 

(The error is not produced in CI process.)

```
$ docker-compose run --rm -w /tmp/ganesha -u ganesha client vendor/bin/phpunit

PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

.................................................E

Time: 14.33 seconds, Memory: 12.00MB

There was 1 error:

1) Ackintosh\Ganesha\Storage\Adapter\PredisRedisTest::incrementAndLoad
Predis\Connection\ConnectionException: Connection refused [tcp://127.0.0.1:6379]

/tmp/ganesha/vendor/predis/predis/src/Connection/AbstractConnection.php:155
/tmp/ganesha/vendor/predis/predis/src/Connection/StreamConnection.php:128
/tmp/ganesha/vendor/predis/predis/src/Connection/StreamConnection.php:178
/tmp/ganesha/vendor/predis/predis/src/Connection/StreamConnection.php:100
/tmp/ganesha/vendor/predis/predis/src/Connection/AbstractConnection.php:81
/tmp/ganesha/vendor/predis/predis/src/Connection/StreamConnection.php:258
/tmp/ganesha/vendor/predis/predis/src/Client.php:215
/tmp/ganesha/tests/Ackintosh/Ganesha/Storage/Adapter/PredisRedisTest.php:13
/tmp/ganesha/tests/Ackintosh/Ganesha/Storage/Adapter/AbstractRedisTest.php:28
```

Comment (ackintosh) [reply tests/Ackintosh/Ganesha/Storage/Adapter/PredisRedisTest.php]: I think we should specify the connection parameter like below:

```diff
-        $r = new \Predis\Client(getenv('GANESHA_EXAMPLE_REDIS') ?: 'localhost');
+        $r = new \Predis\Client([
+                'scheme' => 'tcp',
+                'host' => getenv('GANESHA_EXAMPLE_REDIS') ?: 'localhost',
+                'port' => 6379
+        ]);
```

Comment (b1rdex) [reply tests/Ackintosh/Ganesha/Storage/Adapter/PredisRedisTest.php]: Fixed

**Comment by ackintosh** [comment]
@b1rdex Thanks again. The feature has been released as v0.4.0. 🎉 

https://github.com/ackintosh/ganesha/releases/tag/0.4.0

**Pull request merged**